src/context-items/sanitization.ts

```diff
@@ -13,8 +13,16 @@ export class Sanitization implements ContextItem {
   async run(context: Context, value: any, meta: Meta) {
     const { path, location } = meta;
 
+    const runCustomSanitizer = async () => {
+      const sanitizerValue = this.sanitizer(value, meta);
+      if (sanitizerValue instanceof Promise) {
+        return await sanitizerValue;
+      }
+      return sanitizerValue;
```

Comment:
I think this can simply be

```suggestion
      return Promise.resolve(sanitizerValue);
```

so that we can have the sync or async value wrapped in a promise, ready to be treated in case there's a rejection.